buffer/src/main/java/io/netty/buffer/PooledSlicedByteBuf.java

```diff
@@ -39,20 +39,26 @@ protected PooledSlicedByteBuf newObject(Handle<PooledSlicedByteBuf> handle) {
         }
     };
 
-    static PooledSlicedByteBuf newInstance(AbstractByteBuf buffer, int index, int length, int adjustment) {
-        if (isOutOfBounds(index, length, buffer.capacity())) {
-            throw new IndexOutOfBoundsException(buffer + ".slice(" + index + ", " + length + ')');
+    static PooledSlicedByteBuf newInstance(AbstractByteBuf unwrapped, ByteBuf wrapped,
+                                           int index, int length) {
+        if (isOutOfBounds(index, length, unwrapped.capacity())) {
+            throw new IndexOutOfBoundsException(unwrapped + ".slice(" + index + ", " + length + ')');
         }
```

Comment:
Yes a stativ helper method

> Am 29.07.2016 um 17:20 schrieb Scott Mitchell <notifications@github.com>:
> 
> In buffer/src/main/java/io/netty/buffer/PooledSlicedByteBuf.java:
> 
> > @@ -39,20 +39,26 @@ protected PooledSlicedByteBuf newObject(Handle<PooledSlicedByteBuf> handle) {
> >          }
> >      };
> >  
> > -    static PooledSlicedByteBuf newInstance(AbstractByteBuf buffer, int index, int length, int adjustment) {
> > -        if (isOutOfBounds(index, length, buffer.capacity())) {
> > -            throw new IndexOutOfBoundsException(buffer + ".slice(" + index + ", " + length + ')');
> > +    static PooledSlicedByteBuf newInstance(AbstractByteBuf unwrapped, ByteBuf wrapped,
> > +                                           int index, int length) {
> > +        if (isOutOfBounds(index, length, unwrapped.capacity())) {
> > +            throw new IndexOutOfBoundsException(unwrapped + ".slice(" + index + ", " + length + ')');
> >          }
> can you clarify what you mean? do you mean make a function to share code for the isOutOfBounds/throw?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>